### PR TITLE
Enable create a custom plugin using ts

### PR DESCRIPTION
**Pull request by pnowak**
### Context
Without this improvement we have a TS bug - `Type 'Base' is not a constructor function type.`.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #6107 
2. #6168

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation.

### CODE REVIEW

**Comment by wojciechczerniak** [comment]
Will this be enough to build custom TS plugin? What about other statics that are available through Handsontable class? `Core` for example.

Here is an use case: https://github.com/handsoncode-apps/datasource-connector
Remove custom `handsontable.d.ts` with 3rd party types overrides.

File: handsontable.d.ts [hunk 1]
```diff
@@ -705,26 +705,32 @@ declare namespace Handsontable {
...
     }
 
-    class Base {
+    class BasePlugin {
       pluginName: string;
       pluginsInitializedCallback: any[];
       isPluginsReady: boolean;
       enabled: boolean;
       initialized: boolean;
+      hot: any;
```
Comment (wojciechczerniak) [inline handsontable.d.ts]: Why `any`? Isn't it Handsontable instance?

Comment (pnowak) [reply handsontable.d.ts]: You right, I changed this.

**Comment by jansiegel** [comment]
Why is the plugins' `Base` class renamed to `BasePlugin`, but editors' `Base` is not `BaseEditor`?

**Comment by pnowak** [comment]
Good point. I will change editors' `Base` to`BaseEditor` to keep consistency.

**Comment by zoladkow** [comment]
Guys,

While this PR is slowly progressing (I hope) any ideas how to instantiate EventManager on an external plugin while using typescript right now?

**Pull request merged**